common/component/kafka/metadata.go

```diff
@@ -126,8 +142,8 @@ func (k *Kafka) upgradeMetadata(metadata map[string]string) (map[string]string,
 	}
 
 	// if consumeRetryEnabled is not present, use component default value
-	consumeRetryEnabledVal, consumeRetryEnabledPres := metadata[consumeRetryEnabled]
-	if !consumeRetryEnabledPres || consumeRetryEnabledVal == "" {
+	consumeRetryEnabledVal, consumeRetryEnabledOk := metadata[consumeRetryEnabled]
```

Comment:
Please use [`metadata.GetMetadataProperty`](https://pkg.go.dev/github.com/dapr/kit@v0.13.0/metadata#GetMetadataProperty) which is case-insensitive

And, please use `utils.IsTruthy` to check if the value is truthy

Reply:
updated, but not sure how IsTruthy is applicable here as it does not make reference to if empty string...

Reply:
If it's empty, `IsTruthy` returns false